pkgs/by-name/ps/pshs/package.nix

```diff
@@ -8,38 +8,47 @@
   file,
   qrencode,
   miniupnpc,
+  nix-update-script,
 }:
 
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "pshs";
   version = "0.3.4";
 
   src = fetchFromGitHub {
     owner = "mgorny";
     repo = "pshs";
-    rev = "v${version}";
+    rev = "v${finalAttrs.version}";
     sha256 = "1j8j4r0vsmp6226q6jdgf9bzhx3qk7vdliwaw7f8kcsrkndkg6p4";
   };
 
   nativeBuildInputs = [
     autoreconfHook
     pkg-config
   ];
+
   buildInputs = [
     libevent
     file
     qrencode
     miniupnpc
   ];
 
+  strictDeps = true;
+
   # SSL requires libevent at 2.1 with ssl support
   configureFlags = [ "--disable-ssl" ];
 
+  __structuredAttrs = true;
+
+  passthru.updateScript = nix-update-script { };
```

Comment:
Is this really needed? Updating this package should be trivial for both `nix{,pkgs}-update`.

Reply:
Honestly, it’s never been clear to me exactly where `nixpkgs-update` will update from. I know the main source is Repology, but that it can maybe also do GitHub releases? In this case though the newer releases are just tags and there are no corresponding GitHub releases. This at least lets `maintainers/script/update.nix` work where it doesn’t otherwise.

I asked about best practices [on Matrix](https://matrix.to/#/!kjdutkOsheZdjqYmqp:nixos.org/$UZm4T96zi4OSr7IUL9sq9Rot3olVYz7j50o33R2bU5E?via=lossy.network&via=matrix.org&via=nixos.dev) recently but there wasn’t really any answer. There has been some talk about either making `nix-update-script` the default, or doing a mass addition of it; see https://github.com/NixOS/nixpkgs/pull/325074.